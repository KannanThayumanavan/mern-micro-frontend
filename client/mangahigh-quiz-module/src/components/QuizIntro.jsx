import React from 'mangahigh-libs/react';
import Button from 'mangahigh-components/Button';
import PropTypes from 'prop-types';

const QuizIntro = ({ introHeader, updateProgress }) => (
	<div className="p-3">
		<h5 className="p-3">{introHeader}</h5>		
		<Button
	  	type='success'
	  	buttonLabel='Start Quiz'
	  	buttonOnClick={() => updateProgress('quiz')}
		/>
	</div>
);

QuizIntro.propTypes = {
	introHeader: PropTypes.string.isRequired,
	updateProgress: PropTypes.func.isRequired,
};

export default QuizIntro;