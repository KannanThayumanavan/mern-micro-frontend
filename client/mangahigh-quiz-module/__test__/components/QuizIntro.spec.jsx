import React from 'react';
import { shallow, mount } from 'enzyme';
import QuizIntro from '../../src/components/QuizIntro';

jest.mock('mangahigh-libs/react', () => jest.requireActual('react'),
	{ virtual: true }
);

jest.mock('mangahigh-components/Button', () => 'Button',	
{ virtual: true });

describe('Test QuizIntro Component', () => {
	const mockCallback = jest.fn();
	const props = {
		introHeader: 'test header', 
		updateProgress: mockCallback,
	};
	
	it('should render correctly with required props', () => {
		const renderedModule = shallow(<QuizIntro {...props} />);
    expect(renderedModule).toMatchSnapshot();
	});

	it('should callback when Button clicked', () => {
		const renderedModule = shallow(<QuizIntro {...props} />);								
		// renderedModule.find('Button').simulate('click'); --> line#26 - usual way
		/* line#28 - because remote module(mangahigh-components) actual components are not available 
		*	 https://scriptedalchemy.medium.com/module-federation-how-do-we-create-unit-tests-for-it-bd0d73c999bc	 
		*/
		renderedModule.find('Button').props().buttonOnClick(); 
		expect(mockCallback).toHaveBeenCalledWith('quiz');		
	});
});