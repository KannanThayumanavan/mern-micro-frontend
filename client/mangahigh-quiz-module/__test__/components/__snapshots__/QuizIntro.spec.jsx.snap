// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test QuizIntro Component should render correctly with required props 1`] = `
<div
  className="p-3"
>
  <h5
    className="p-3"
  >
    test header
  </h5>
  <Button
    buttonLabel="Start Quiz"
    buttonOnClick={[Function]}
    type="success"
  />
</div>
`;
